Neos.Neos/Classes/Controller/Service/AssetProxiesController.php

```diff
@@ -0,0 +1,167 @@
+use Neos\Media\Domain\Service\AssetSourceService;
+use Neos\Media\Domain\Repository\TagRepository;
+use Neos\Media\Exception\AssetSourceServiceException;
+use Neos\Neos\Controller\BackendUserTranslationTrait;
+use Neos\Neos\View\Service\AssetJsonView;
+
+/**
+ * Rudimentary REST service for assets
+ *
+ * @Flow\Scope("singleton")
+ */
+class AssetProxiesController extends ActionController
+{
+    use BackendUserTranslationTrait;
+
+    /**
+     * @Flow\Inject
+     * @var AssetRepository
+     */
+    protected $assetRepository;
+
+    /**
+     * @Flow\Inject
+     * @var AssetSourceService
+     */
+    protected $assetSourceService;
+
+    /**
+     * @Flow\Inject
+     * @var TagRepository
+     */
+    protected $tagRepository;
+
+    /**
+     * @Flow\InjectConfiguration(package="Neos.Media", path="asyncThumbnails")
+     * @var boolean
+     */
+    protected $asyncThumbnails;
+
+    /**
+     * @var array
+     */
+    protected $viewFormatToObjectNameMap = array(
+        'html' => TemplateView::class,
+        'json' => AssetJsonView::class
+    );
+
+    /**
+     * A list of IANA media types which are supported by this controller
+     *
+     * @var array
+     * @see http://www.iana.org/assignments/media-types/index.html
+     */
+    protected $supportedMediaTypes = array(
+        'text/html',
+        'application/json'
+    );
+
+    /**
+     * @param ViewInterface $view
+     * @return void
+     */
+    public function initializeView(ViewInterface $view)
+    {
+        $view->assign('asyncThumbnails', $this->asyncThumbnails);
+    }
+
+    /**
+     * Shows a list of assets
+     *
+     * @param string $searchTerm An optional search term used for filtering the list of assets
+     * @param string $assetSourceIdentifier If specified, results are only from the given asset source
+     * @param int $limitPerAssetSource The maximum number of results shown per asset source
+     * @return void
+     */
+    public function indexAction(string $searchTerm = '', string $assetSourceIdentifier = '', int $limitPerAssetSource = 10): void
+    {
+        $assetSources = $this->assetSourceService->getAssetsSources();
+        $assetProxies = [];
+        foreach ($assetSources as $assetSource) {
+            if ($assetSourceIdentifier !== '' && $assetSource->getIdentifier() !== $assetSourceIdentifier) {
+                continue;
+            }
+            try {
+                $assetProxyRepository = $assetSource->getAssetProxyRepository();
+                $assetProxyQueryResult = $assetProxyRepository->findBySearchTerm($searchTerm);
+                $addedResults = 0;
+                while ($assetProxyQueryResult->valid() && $addedResults < $limitPerAssetSource) {
+                    $assetProxies[] = $assetProxyQueryResult->current();
+                    $assetProxyQueryResult->next();
+                    $addedResults++;
+                }
+            } catch (\Exception $exception) {
```

Comment:
catching the base exception is quite dangerous. Is that really required here?

Reply:
We could log the exception, but there is no meaningful way to display the error to the user at this point.